File: user-service/src/main/java/com/animal/userservice/data/model/Personality.java
package com.animal.userservice.data.model;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class Personality {
    private Level sociability;
    private Level activity;
    private Level stability;
    private Level patience;
    private Level motivation;
    private Level adaptability;
    private Level communication;
}

File: ui/src/components/userProfile/Personality.js
import React from "react";
import Level from "../Level";

export default function Personality({data, handleOnChange}){
    const branch = "personality";
    const personalityList = [
        "Sociability",
        "Activity",
        "Stability",
        "Patience",
        "Motivation",
        "Adaptability",
        "Communication"
    ];

    const personalityInputs = personalityList.map(p => {
        const lowerCasedP = p.toLowerCase();
        return <Level 
                    key={lowerCasedP}
                    purpose={p} 
                    name={lowerCasedP}
                    data={data[lowerCasedP]}
                    handleLevelChange={handleOnChange} 
                    branch={branch}
                />; 
    });

    return (
        <div>
            <h3>Personality</h3>
            {personalityInputs}
        </div>
    )
}